### DISCUSSION TITLE
Where snippets are defined?

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [Beaglefoot/awk-language-server#42] [Q&A] [answered]
I wanna add more ones but can't find what file to modify.

**Comment by Beaglefoot** [comment]
You are probably looking for `server/docs.json` file, which is generated with `server/scripts/build_completion_list.awk` script from linux manual page on GAWK. The contents are not meant for manual editing though.

What snippets are you missing?

**Comment by EmilyGraceSeville7cf** [comment]
> What snippets are you missing?

Snippets for compound commands, like `if`. I will send PR tomorrow to add VS Code-specific snippets. I want them to be similar to those Bash IDE extension has to have similar user experience.

As a sidenote, Bash IDE extension [serves](https://github.com/bash-lsp/bash-language-server/blob/20c1b14f78b68bc71697a7a3fb9c4b38ce419e6a/server/src/snippets.ts) snippets from their LSP. I thought your extension does it in the same manner.

**Reply by Beaglefoot** [reply, marked as answer]
> As a sidenote, Bash IDE extension [serves](https://github.com/bash-lsp/bash-language-server/blob/20c1b14f78b68bc71697a7a3fb9c4b38ce419e6a/server/src/snippets.ts) snippets from their LSP. I thought your extension does it in the same manner.

Now it does. You can continue adding more [here](https://github.com/Beaglefoot/awk-language-server/blob/master/server/snippets.json).
